General updates (#810)
maticnetwork/matic-docs: docs/develop/ethereum-polygon/pos/using-sdk/erc1155.md

---

### approveALL

[approveAll method](https://maticnetwork.github.io/matic.js/docs/pos/erc1155/approve-all/)

> This is a normal ERC1155 approval so that **_ERC1155Predicate_** can call **_transferFrom_** function. Polygon POS client exposes [**_approveERC1155ForDeposit_** ](https://github.com/maticnetwork/matic.js/blob/4bf4fa9438d56c9b5c282f456aa2c24f6ff6083d/src/index.ts#L231) method to make this call.

Reviewer: ```suggestion
This is a normal ERC1155 approval so that `ERC1155Predicate` can call `transferFrom` function. The Polygon `POSClient` exposes the [`approveERC1155ForDeposit`](https://github.com/maticnetwork/matic.js/blob/4bf4fa9438d56c9b5c282f456aa2c24f6ff6083d/src/index.ts#L231) method to make this call.
```

```jsx
const erc1155RootToken = posClient.erc1155(<root token address>,true);
const approveResult = await erc1155RootToken.approveAll();
const txHash = await approveResult.getTransactionHash();
const txReceipt = await approveResult.getReceipt();

```

### Deposit

> [Deposit method](https://maticnetwork.github.io/matic.js/docs/pos/erc1155/deposit/) can be done by calling **_depositFor_** on RootChainManager contract. Note that token needs to be mapped and approved for transfer beforehand. Once tokens are transferred deposit proceeds using StateSync mechanism. Polygon POS client exposes [**_depositSingleERC1155ForUser_**](https://github.com/maticnetwork/matic.js/blob/4bf4fa9438d56c9b5c282f456aa2c24f6ff6083d/src/index.ts#L245) & [**_depositBatchERC1155ForUser_**](https://github.com/maticnetwork/matic.js/blob/4bf4fa9438d56c9b5c282f456aa2c24f6ff6083d/src/index.ts#L259) method to make this call.

Reviewer: ```suggestion
[Deposit method](https://maticnetwork.github.io/matic.js/docs/pos/erc1155/deposit/) can be done by calling `depositFor` on the `RootChainManager` contract. Note that token needs to be mapped and approved for transfer beforehand. Once tokens are transferred deposit proceeds using **_StateSync_** mechanism. The Polygon `POSClient` exposes the [`depositSingleERC1155ForUser`](https://github.com/maticnetwork/matic.js/blob/4bf4fa9438d56c9b5c282f456aa2c24f6ff6083d/src/index.ts#L245) & [`depositBatchERC1155ForUser`](https://github.com/maticnetwork/matic.js/blob/4bf4fa9438d56c9b5c282f456aa2c24f6ff6083d/src/index.ts#L259) methods to make this call.
```

```jsx
const erc1155RootToken = posClient.erc1155(<root token address>, true);

const result = await erc1155RootToken.deposit({
    amount: 1,
    tokenId: '123',
    userAddress: <from address>,
    data: '0x5465737445524331313535', // data is optional
});

const txHash = await result.getTransactionHash();
const txReceipt = await result.getReceipt();
```

> **_deposit_** function of **_ChildToken_** is called by the **_ChildChainManager._** Tokens should be minted when this call is made.

Reviewer: ```suggestion
The `deposit` function of `ChildToken` is called by the `ChildChainManager`. Tokens should be minted when this call is made.
```

> User can call **_withdraw_** function of **_ChildToken_** contract. This function should burn the tokens. Polygon POS client exposes **_burnSingleERC1155_** & **_burnBatchERC1155_** method to make this call.

Reviewer: ```suggestion
User can call the `withdraw` function of `ChildToken` contract. This function should burn the tokens. The Polygon `POSClient` exposes the `burnSingleERC1155` & `burnBatchERC1155` methods to make this call.
```

```jsx
const erc1155Token = posClient.erc1155(<child token address>);
const result = await erc1155Token.withdrawStart(<token id>,<amount>);
const txHash = await result.getTransactionHash();
const txReceipt = await result.getReceipt();
```

**Store the transaction hash for this call and use it while generating burn proof.**

### Exit

[WithdrawExit method](https://maticnetwork.github.io/matic.js/docs/pos/erc1155/withdraw-exit/)

> Once the **_checkpoint_** has been **_submitted_** for the block containing burn transaction, user should call the **_exit_** function of **_RootChainManager_** contract and submit the proof of burn. Upon submitting valid proof tokens are transferred to the user. Polygon POS client exposes **_exitBatchERC1155_** & **_exitSingleERC1155_** method to make this call. This function can be called only after the checkpoint is included in the main chain. The checkpoint inclusion can be tracked by following this [guide](/docs/develop/ethereum-polygon/pos/deposit-withdraw-event-pos#checkpoint-events).

Reviewer: ```suggestion
Once the `checkpoint` has been **_submitted_** for the block containing burn transaction, a user should call the `exit` function of the `RootChainManager` contract and submit the proof of burn. Upon submitting valid proof tokens are transferred to the user. Polygon POS client exposes `exitBatchERC1155` & `exitSingleERC1155` method to make this call. This function can be called only after the checkpoint is included in the main chain. The checkpoint inclusion can be tracked by following this [guide](/docs/develop/ethereum-polygon/pos/deposit-withdraw-event-pos#checkpoint-events).
```
